oshdb/src/main/java/org/heigit/ohsome/oshdb/osm/OSM.java

```diff
@@ -42,6 +55,21 @@ public static OSMRelation relation(final long id, final int version, final long
     return new Relation(id, version, timestamp, changeset, userId, tags, members);
   }
 
+  public static OSMRelation relation(final long id, final int version, final long timestamp,
+      final long changeset, final int userId, final List<OSHDBTag> tags, final OSMMember[] members) {
+    return new Relation(id, version, timestamp, changeset, userId, toKV(tags), members);
+  }
+
+  private static int[] toKV(List<OSHDBTag> tags) {
+    var kv = new int[tags.size() * 2];
+    for (var i=0; i < tags.size(); i++) {
```

Comment:
(if we had a list/collection i would prefer to iterate over the `tags`list in order to save the index computation)

Reply:
As kv is twice the size of tags. We need some sort of index to calculate the k and v position
```
 kv[i * 2] = key
 kv[i * 2 + 1] = value
 ```
 
What do you think about this variant:
 
 ```
 var i = 0;
 for (var tag : tags) {
  kv[i++] = tag.getKey();
  kv[i++] = tag.getValue();
 }
 ```

Reply:
i know that the result structure is twice as long, but as we access it only in ascending order it would be safe to just append the 2 new output entries (per input entry) to the result.
(if we had a `Collection` type instead of an array).
as the _array_ is also used in the consuming type we should probably keep it.

Reply:
and yes, the second variant looks nicer to me
:)